src/diffpy/srmise/basefunction.py

```diff
@@ -114,14 +115,6 @@ def __init__(
             if not isinstance(p, str):
                 emsg = "Argument parameterdict's keys must be strings."
                 raise ValueError(emsg)
-        vals = self.parameterdict.values()
-        vals.sort()
```

Comment:
`val.sort()` no longer works in Python 3.12 since the line above `parameterdict.values()` does not return a list.